### ISSUE TITLE
Parallel prefix sum

### ISSUE THREAD

**Issue by corporateshark** [taskflow/taskflow#347]
An implementation of any modern parallel prefix sum algorithm would be a great addition to the library.

```
tf::Task task1 = taskflow.prefix_sum_exclusive(
  first, last, [] (auto& i) -> uint { return first[i]; }    
);
tf::Task task2 = taskflow.prefix_sum_inclusive(
  first, last, [] (auto& i) -> uint { return first[i]; }    
);
```

**Comment by tsung-wei-huang** [comment]
This is a good idea. Should it be a binary operator instead of a unary operator?

**Comment by corporateshark** [comment]
I'm thinking, maybe it can be just a templated function operating on a container of ints/floats? No lambda will be required this way.

**Comment by tsung-wei-huang** [comment]
Has been implemented in the current branch - will be included in the next release 3.6.0
